src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs

```diff
@@ -140,10 +137,11 @@ internal async Task<OpenApiSchema> GetOrCreateUnresolvedSchemaAsync(OpenApiDocum
     internal async Task<IOpenApiSchema> GetOrCreateSchemaAsync(OpenApiDocument document, Type type, IServiceProvider scopedServiceProvider, IOpenApiSchemaTransformer[] schemaTransformers, ApiParameterDescription? parameterDescription = null, CancellationToken cancellationToken = default)
     {
         var schema = await GetOrCreateUnresolvedSchemaAsync(document, type, scopedServiceProvider, schemaTransformers, parameterDescription, cancellationToken);
-        return ResolveReferenceForSchema(document, schema);
+        var baseSchemaId = optionsMonitor.Get(documentName).CreateSchemaReferenceId(_jsonSerializerOptions.GetTypeInfo(type));
```

Comment:
Is this expensive? Do we care?

Reply:
Yeah, I think we can probably cache the ID here. We'd expect `GetOrCreateSchemaAsync` to be called on the same types repeatedly (e.g. it appears in the request and in the response) so we can cache for subsequent calls.